### PR TITLE
ref(processor): Remove reservoir from the state

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR removes the `ReservoirEvaluator` from the state. The PR is part of a series of smaller PRs that remove each item from the state individually.

#skip-changelog

### CODE REVIEW

File: relay-server/src/services/processor.rs [hunk 1]
```diff
@@ -1893,6 +1895,34 @@ impl EnvelopeProcessorService {
                 .parametrize_dsc_transaction(&sampling_state.config.tx_name_rules);
         }
 
+        macro_rules! call {
+            ($fn_name:ident ( $($args:expr),* )) => {{
```
Comment (iambriccardo) [inline relay-server/src/services/processor.rs]: This solution for params was the easiest i could think of. I am not fond of using macros but for this use case I feel like it's pretty okay.

Comment (loewenheim) [reply relay-server/src/services/processor.rs]: Isn't this macro the same as `run`, except it also allows specifying arguments? I think the two macros can be unified.

Comment (iambriccardo) [reply relay-server/src/services/processor.rs]: Yes, we could, I am currently discussing offline which approach we want to take.

File: relay-server/src/services/processor.rs [hunk 2]
```diff
@@ -1894,7 +1896,7 @@ impl EnvelopeProcessorService {
         }
 
         macro_rules! run {
-            ($fn:ident) => {{
+            ($fn_name:ident , $($args:expr),*) => {{
```
Comment (jjbayer) [inline relay-server/src/services/processor.rs]: to prevent trailing commas

```suggestion
            ($fn_name:ident $(, $args:expr)*) => {{
```

**Review by jjbayer — APPROVED** [review]

**Pull request merged**